api/src/socketio.ts

```diff
@@ -41,38 +41,41 @@ export class Socketio implements OnGatewayInit {
                 return;
             }
 
-            // Broadcast CMS specific documents to CMS group rooms
-            if (this.cmsDocTypes.includes(update.type)) {
-                // Change documents referencing a non-group document
-                if (update.memberOf) {
-                    server
-                        .to(update.memberOf.map((group: Uuid) => `cms-${group}`))
-                        .emit("data", [update]);
-                    return;
+            // We are using a socket.io room per document type per group. Change documents are broadcasted to the document-group rooms of the documents they reference.
+            // Content documents are broadcasted to their parent document-group rooms.
+
+            // Non-group documents
+            if (update.memberOf) {
+                let docType = update.type;
+                if (update.type == "change" && update.changes) {
+                    if (update.changes.type == "content") {
+                        docType = update.changes.parentType;
+                    } else {
+                        docType = update.changes.type;
+                    }
+                    docType = "cms-" + docType; // Prepend "cms-" to the docType for (CMS only) change documents. This is needed to be able to allow the CMS to specifically subscribe to change documents.
+                } else if (update.type == "content") {
+                    docType = update.parentType;
                 }
 
-                // Group documents and Change documents referencing a group document
-                if (
-                    update.acl &&
-                    (update.type == DocType.Group || update.docType == DocType.Group)
-                ) {
-                    server
-                        .to(`cms-${update.docId ? update.docId : update._id}`)
-                        .emit("data", [update]);
-                    return;
-                }
-
-                // TODO: Add error logging provider
+                server
+                    .to(update.memberOf.map((group) => `${docType}-${group}`))
+                    .emit("data", [update]);
                 return;
             }
 
-            // All other valid documents - broadcast to App and CMS group rooms
-            if (this.appDocTypes.includes(update.type)) {
-                server
-                    .to(update.memberOf.map((group) => `cms-${group}`))
-                    .to(update.memberOf.map((group) => `app-${group}`))
-                    .emit("data", [update]);
+            // Group documents
+            if (update.acl) {
+                // If the document is a "change" document, the group id is stored in the docId property
+                let docType = update.type;
+                let groupId = update._id;
 
+                if (update.type == "change" && update.changes) {
+                    docType = update.changes.type;
+                    groupId = update.docId;
+                }
```

Comment:
I'm a little confused why we're testing for change here but perhaps you can take me through this method to make me understand it better.